api/services/DashboardDataMassifService.js

```diff
@@ -0,0 +1,211 @@
+const CREATE_MATERIALIZED_VIEW = `
+  CREATE OR REPLACE MATERIALIZED VIEW MassifInfoView AS
+  SELECT m.id as id_massif,
+  c.id as id_cave,
+  n.name as name_cave,
+  c.depth as depth_cave,
+  c.length as length_cave,
+  c.is_diving as is_diving_cave,
+  e.id as id_entrance,
+  e.is_sensitive as is_sensitive_entrance
+  FROM t_massif m 
+  JOIN t_entrance e ON ST_Contains(ST_SetSRID(m.geog_polygon::geometry, 4326), ST_SetSRID(ST_MakePoint(e.longitude, e.latitude), 4326))
+  JOIN t_cave c ON e.id_cave = c.id 
+  JOIN t_name n ON n.id_cave = c.id 
+  WHERE n.is_main = true 
+  AND c.is_deleted = false
+  AND e.is_deleted = false
+  AND m.is_deleted = false;
+`;
+
+const GET_NB_CAVES = `
+SELECT COUNT(*) as nb_caves
+FROM (
+	SELECT DISTINCT id_cave 
+	FROM MassifInfoView
+  WHERE id_massif = $1) as tmp
+`;
+
+const FIND_CAVE_WITH_MAX_DEPTH_IN_MASSIF = `
+  SELECT id_cave, name_cave, depth_cave
+  FROM MassifInfoView
+  WHERE depth_cave IN (
+    SELECT MAX(depth_cave)
+    FROM MassifInfoView
+    WHERE id_massif = $1
+    )
+  AND id_massif = $1
+`;
+
+const FIND_CAVE_WITH_MAX_LENGTH_IN_MASSIF = `
+  SELECT id_cave, name_cave, length_cave
+  FROM MassifInfoView
+  WHERE length_cave IN (
```

Comment:
Si le but est de ne trouver que 1 seul cave avec le max length il ne faudrait pas un `=` au lieu d'un `IN` ?